_posts/2022-01-28-data-prepper-1.2.1-performance-testing.md

```diff
@@ -0,0 +1,102 @@
+| HTTP Source Thread   |            4 |        4 |
+| Max Connection Count |        2,000 |    n/a * |
+| Request Timeout      |       10,000 |    n/a * |
+| Buffer Size          |    2,000,000 |    n/a * |
+| Batch Size           |        5,000 |    5,000 |
+| Worker Thread        |           12 |       12 |
+| SSL                  |     Disabled | Disabled |
+
+_[*] Note, Max Connection Count, Request Timeout, Buffer Size are not configurable with Logstash 7.13.2_
+
+## Data Prepper Scaling
+
+In this simulation clients are ramped up from 1 to 60 over one hour to measure the impact concurrent clients have on performance.
+
+![Data Prepper Throughput Increasing Users](/assets/media/blog-images/2022-01-28-data-prepper-1.2.1-performance-testing/Graph-Throughput-Increasing-Users.png){: .img-fluid}
+
+As the number of clients increases Data Preppers throughput remains constant, processing an average of 10,473 MB/m.
+
+# Limitation of Testing
+
+It’s important to note that Data Prepper 1.2.1 and Logstash 7.13.2 support different feature sets, and our performance test suite is targeted at common functionality. As Data Prepper adds more processors and sources, test cases will be added. The Data Prepper team will continue update the community with performance benchmarks.
+
+## Candidates for future performance testing scenarios and improvements
+
+![Environment Reference](/assets/media/blog-images/2022-01-28-data-prepper-1.2.1-performance-testing/Environment-Reference.png){: .img-fluid}
+
+In a real-world deployment, if Data Prepper is unable to keep pace with the logs generated by the source application. Data Prepper will become a bottleneck causing backpressure on the source application. In the future performance tests will be enhanced to simulate backpressure and measure the impact.
+
+The scope of our initial performance testing scenarios was focused on a common scenario, [http source](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/http-source), [grok processor](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/grok-prepper), and an [OpenSearch sink](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/opensearch). As new features are added to Data Prepper in upcoming releases, performance testing simulations will be added to cover core functionality.
+
+# Running Data Prepper Performance tests
+
+Performance suite source code is available on [GitHub](https://github.com/sbayer55/gatling-tests/). To run the full test suite execute `./gradlew –rerun-tasks gatlingRun -Dhost=<target url>`. After all tests have completed HTML reports will be created in `./build/reports/gatling/<simulation-name>-<unix-timestamp>/index.html`. Further instructions on running performance tests and [Gatling](https://gatling.io/) are available in the repository [readme](https://github.com/sbayer55/gatling-tests/blob/main/README.md).
```

Comment:
`https://github.com/sbayer55/gatling-tests/` should not be on a personal github account.